Gems/LyShine/Code/Editor/RulerWidget.cpp

```diff
@@ -61,6 +61,14 @@ void RulerWidget::paintEvent([[maybe_unused]] QPaintEvent* event)
     float scale = translationAndScale.scale;
     float translation = (m_orientation == Orientation::Horizontal) ? translationAndScale.translation.GetX() : translationAndScale.translation.GetY();
 
+    // Convert back to qt widget coords for painting
+    float dpiScaleFactor = m_editorWindow->GetViewport()->WidgetToViewportFactor();
+    if (dpiScaleFactor != 0)
```

Comment:
Is this check necessary? If this can be 0, do we need to anything in other cases that check this, like your code in `ViewportHelpers`?

Reply:
This check is here to avoid a potential divide by zero since it's doing a division by the scale factor here rather than a multiplication. However, it is probably safe to say that it won't ever be 0.

Reply:
Nit: 0.0f instead of 0 since `dpiScaleFactor` is float.